mlflow/transformers/__init__.py

```diff
@@ -193,15 +197,6 @@ def get_default_pip_requirements(model) -> List[str]:
         ``transformers`` flavor. Calls to :py:func:`save_model()` and :py:func:`log_model()`
         produce a pip environment that contain these requirements at a minimum.
     """
-
-    from transformers import FlaxPreTrainedModel, PreTrainedModel, TFPreTrainedModel
-
-    if not isinstance(model, (TFPreTrainedModel, FlaxPreTrainedModel, PreTrainedModel)):
```

Comment:
Do we want to keep this type check and include peft model? If we remove this check, is the error stack trace clear?

Reply:
Great point! There were some anecdotes that our custom validation confused users, so ideally we can let Transformers to handle that. That said, it seems for this one the error message for passing wrong type is quite confusing. Will add back this validation (not in `get_default_pip_requirements` tho:p)

![Screenshot 2024-02-15 at 8 53 43](https://github.com/mlflow/mlflow/assets/31463517/6858f21c-28e5-4bbb-a847-a188bd1d79ce)

Reply:
yea the HF error message is leading me to nowhere... thanks Yuki!